jps-plugin/src/org/intellij/erlang/jps/builder/ErlangBuilder.java

```diff
@@ -407,32 +370,14 @@ private static void addModuleToCodePath(@NotNull GeneralCommandLine commandLine,
     }
   }
 
-  private static void markDirtyNodes(@NotNull List<Node> dirtyModules,
-                                     @NotNull GraphGenerator<Node> graph) {
-    for (Node node : dirtyModules) {
-      if (node != null) {
-        markDirtyNodes(node, graph);
-      }
-    }
-  }
-
-  private static void markDirtyNodes(@NotNull Node node, @NotNull GraphGenerator<Node> graph) {
-    if (node.myDirty) return;
-    node.myDirty = true;
-    Iterator<Node> childIterator = graph.getOut(node);
-    while (childIterator.hasNext()) {
-      markDirtyNodes(childIterator.next(), graph);
-    }
-  }
-
   @NotNull
-  private static List<File> getOutputErlangModuleFiles(@NotNull List<String> erlangModulePathsToCompile,
+  private static List<File> getOutputErlangModuleFiles(@NotNull Collection<String> erlangModulePathsToCompile,
                                                        @NotNull final File outputDirectory) {
```

Comment:
Sure, getBeams would also do just fine. Also, there's no reason to have these paths named 'erlangModulePathsToCompile' - it could be just 'erlPaths', or 'erls' - it's way simpler and more readable without any extraneous information.